# 1. Imports

In [51]:
import pandas as pd

pd.set_option('display.max_colwidth',100)

# 2. Cargar el CSV crudo

Ajusta el nombre del archivo según el restaurante que estés revisando. Este notebook corre **una vez por cada CSV nuevo** que descargues de Apify, antes de moverlo a `data/raw/`para la limpieza masiva. 

In [52]:
df = pd.read_csv('../data/raw/parrilla_punta_del_este_reviews_raw.csv')
print(f'shape {df.shape}')

shape (596, 77)


# 3. Revisar columnas disponibles

In [53]:
print(f'Total de columnas {len(df.columns)}')

for c in df.columns: 
    print("-", c)

Total de columnas 77
- address
- businessProfileId
- categories/0
- categoryName
- cid
- city
- countryCode
- fid
- hotelStars
- imageUrl
- isAdvertisement
- isLocalGuide
- kgmid
- language
- likesCount
- location/lat
- location/lng
- name
- neighborhood
- originalLanguage
- permanentlyClosed
- placeId
- postalCode
- price
- publishAt
- publishedAtDate
- rating
- responseFromOwnerDate
- responseFromOwnerText
- reviewContext/Accesibilidad en silla de ruedas
- reviewContext/Espacios para estacionar
- reviewContext/Estacionamiento
- reviewContext/Eventos especiales
- reviewContext/Nivel de ruido
- reviewContext/Opciones de estacionamiento
- reviewContext/Platos recomendados
- reviewContext/Platos vegetarianos
- reviewContext/Precio por persona
- reviewContext/Reservas
- reviewContext/Tamaño del grupo
- reviewContext/Tiempo de espera
- reviewContext/Tipo de asiento
- reviewContext/Tipo de comida
- reviewContext/Tipo de pedido
- reviewDetailedRating/Ambiente
- reviewDetailedRating/Comida
- 

# 4 Completitud de las columnas clave

Revisamos qué % de filas tiene dato real en las columnas que nos interesan para el análisis. 

In [54]:
columnas_interes = [
    'text', 'stars', 'rating', 'publishedAtDate', 'language',
    'originalLanguage', 'reviewDetailedRating/Comida', 'reviewDetailedRating/Servicio',
    'reviewDetailedRating/Ambiente', 'reviewContext/Tiempo de espera',
    'responseFromOwnerText', 'likesCount', 'reviewerNumberOfReviews',
    'isLocalGuide', 'title'
]

for c in columnas_interes: 
    if c in df.columns: 
        nn = df[c].notna().sum()
        print(f'{c:45s} -> no-nulos {nn}/{len(df)}')
    else: 
        print(f'{c:45s} -> columna no encontrada')

text                                          -> no-nulos 506/596
stars                                         -> no-nulos 596/596
rating                                        -> no-nulos 0/596
publishedAtDate                               -> no-nulos 596/596
language                                      -> no-nulos 596/596
originalLanguage                              -> no-nulos 506/596
reviewDetailedRating/Comida                   -> no-nulos 97/596
reviewDetailedRating/Servicio                 -> no-nulos 98/596
reviewDetailedRating/Ambiente                 -> no-nulos 98/596
reviewContext/Tiempo de espera                -> no-nulos 6/596
responseFromOwnerText                         -> no-nulos 0/596
likesCount                                    -> no-nulos 596/596
reviewerNumberOfReviews                       -> no-nulos 596/596
isLocalGuide                                  -> no-nulos 596/596
title                                         -> no-nulos 596/596


# 5. Verificar cuál columna tiene el rating real


In [55]:
print('stars values_counts:')
print(df['stars'].value_counts())
print()
print('rating no-nulos:', df['rating'].notna().sum() if 'rating' in df.columns else 'columna no existe')

stars values_counts:
stars
5    336
4    142
3     78
2     21
1     19
Name: count, dtype: int64

rating no-nulos: 0


# 6. Identificar el restaurante y validar que es un solo local

In [56]:
print(df['title'].unique())

<ArrowStringArray>
['Parrillada Punta Del Este']
Length: 1, dtype: str


# 7. Muestra de reseñas con texto

Lectura rápida y manual de algunas reseñas reales para tener contexto antes de limpiar/modelar

In [57]:
df[df['text'].notna()][['text','stars']].sample(5,random_state=42)

,text,stars
173,"No me gusto,podrían mejorar la parrillada estaba fría y el servicio malo",1
274,"Buena carne, precios módicos",4
491,Excelente lugar,5
72,"Buena atención, buena comida, asistí a un evento en la planta alta... Amplio y espacioso 👍",4
452,Rica parrillada,5


# 8. Análisis

Rukito Grill&drinks - La Alborada: 
- Shape: 1000 filas x 81 columnas
- Columna de rating real: 1000 
- Reseñas con texto: 1000/1000
- Sin problemas de codificación o columnas faltantes detectados


Casa Res | Steak House (Mall del Sol): 
- Shape: 672 filas x 84 columnas
- Columna de rating real: 672 
- Reseñas con texto :  579/672
- Sin problemas de codificación o columnas faltantes detectados


Don Parrilla Steak House - Urdesa:  
- Shape: 272 filas x 84 columnas 
- Columna de rating real: 272 
- Reseñas con texto : 141/272
- Sin problemas de codificación o columnas faltantes detectados

Parrillada Restaurant El Dorado Sauces 3:  
- Shape: 740 filas x  81 columnas  
- Columna de rating real: 740  
- Reseñas con texto :  585/740
- Sin problemas de codificación o columnas faltantes detectados

La Casa del Tomahawk9:  
- Shape:  1000 filas x 99 columnas 
- Columna de rating real: 1000 
- Reseñas con texto :  1000/1000
- Sin problemas de codificación o columnas faltantes detectados

La Parrilla Del Ñato - Urdesa:  
- Shape: 1000 filas x 82 columnas 
- Columna de rating real: 1000 
- Reseñas con texto :  1000/1000
- Sin problemas de codificación o columnas faltantes detectados

MoroGrill - C.C. Las Terrazas:  
- Shape:  572 filas x 84 columnas 
- Columna de rating real: 572 
- Reseñas con texto :  464/572
- Sin problemas de codificación o columnas faltantes detectados

Parrillada Punta Del Este:  
- Shape: 596 filas x 77  columnas 
- Columna de rating real: 596
- Reseñas con texto :  506/596
- Sin problemas de codificación o columnas faltantes detectados